internal/cli/main.go

```diff
@@ -632,26 +632,38 @@ func logger(args []string) ([]string, hclog.Logger, io.Writer, error) {
 	// Process arguments looking for `-v` flags to control the log level.
 	// This overrides whatever the env var set.
 	var outArgs []string
+	// We use this to track if flags are given after a --, indicating they should
+	// be passed through to a secondary command; currently, this is largely to
+	// ensure that the `-v(vv)` flags are passed through to be parsed later;
+	// otherwise we'd swallow them in the below case statement
+	hasDoubleHypen := false
 	for _, arg := range args {
 		if len(arg) != 0 && arg[0] != '-' {
 			outArgs = append(outArgs, arg)
 			continue
 		}
+		if arg == "--" {
+			hasDoubleHypen = true
```

Comment:
When this happens, I'd stop the loop and do something like this:

```go
if arg == "--" {
  outArgs = append(outArgs, args[i:]...)
  break
}
```

Then you could also get rid the below if statement.